### PR TITLE
feat: add DeviceDiscoveryRequest model

### PR DESCRIPTION
**Pull request by instpe**
feat: add DeviceDiscoveryRequest model

### CODE REVIEW

File: mdtp-common/src/main/java/iot/github/protocol/mdtp/common/model/AbstractMessageBody.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package iot.github.protocol.mdtp.common.model;
```
Comment (hezhangjian) [inline mdtp-common/src/main/java/iot/github/protocol/mdtp/common/model/AbstractMessageBody.java]:
```suggestion
package io.github.protocol.mdtp.common.model;
```

File: mdtp-common/src/main/java/io/github/protocol/mdtp/common/model/AbstractMessageBody.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.util.UUID;
+
+@Data
+public abstract class AbstractMessageBody {
+    private short messageBodyHeader;
+
+    public void setMessageBodyHeader(MessageType messageType, ServiceGroup serviceGroup, DiscoveryServiceCode serviceCode) {
+        this.messageBodyHeader = 0;
+
```
Comment (hezhangjian) [inline mdtp-common/src/main/java/io/github/protocol/mdtp/common/model/AbstractMessageBody.java]: this line change is redundant

File: mdtp-common/src/test/java/io/github/protocol/mdtp/common/DeviceDiscoveryResponseTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DeviceDiscoveryResponseTest {
+
+    private DeviceDiscoveryResponse response;
+
+    @BeforeEach
+    void setUp() {
+        // 构造一个 DeviceDiscoveryResponse 对象，使用一些合理的测试数据
```
Comment (hezhangjian) [inline mdtp-common/src/test/java/io/github/protocol/mdtp/common/DeviceDiscoveryResponseTest.java]: remove this annotation

File: pom.xml [hunk 4]
```diff
@@ -28,6 +28,7 @@
         <lombok.version>1.18.34</lombok.version>
         <netty.version>4.1.112.Final</netty.version>
         <slf4j.version>2.0.16</slf4j.version>
+        <mockito.version>5.11.0</mockito.version>
```
Comment (hezhangjian) [inline pom.xml]: sort by alphabet

File: mdtp-common/src/main/java/iot/github/protocol/mdtp/common/model/AbstractMessageBody.java (hunk 1 above)
Comment (instpe) [reply mdtp-common/src/main/java/iot/github/protocol/mdtp/common/model/AbstractMessageBody.java]: Agree

File: mdtp-common/src/main/java/io/github/protocol/mdtp/common/model/AbstractMessageBody.java (hunk 2 above)
Comment (instpe) [reply mdtp-common/src/main/java/io/github/protocol/mdtp/common/model/AbstractMessageBody.java]: Agree

**Review by hezhangjian — APPROVED** [review]

**Pull request merged**